File: domain/src/main/java/io/prizy/domain/contest/ports/ContestTemplateCompiler.java
package io.prizy.domain.contest.ports;

import io.prizy.domain.contest.model.Contest;

/**
 *  @author Nidhal Dogga
 *  @created 4/25/2022 10:00 PM
 */

public interface ContestTemplateCompiler {
  String rulesTemplate(Contest contest);
}

File: domain/src/main/kotlin/io/prizy/domain/contest/ports/ContestTemplateCompiler.kt
package io.prizy.domain.contest.ports

import io.prizy.domain.contest.model.Contest

/**
 *  @author Nidhal Dogga
 *  @created 4/25/2022 10:00 PM
 */

interface ContestTemplateCompiler {
  suspend fun rulesTemplate(contest: Contest): String
}